StoneMasonKarel.java

```diff
@@ -1,17 +1,79 @@
-/*
- * File: StoneMasonKarel.java
- * --------------------------
- * The StoneMasonKarel subclass as it appears here does nothing.
- * When you finish writing it, it should solve the "repair the quad"
- * problem from Assignment 1.  In addition to editing the program,
- * you should be sure to edit this comment so that it no longer
- * indicates that the program does nothing.
- */
-
-import stanford.karel.*;
-
-public class StoneMasonKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: StoneMasonKarel.java
+ * --------------------------
+ * The StoneMasonKarel subclass as it appears here does nothing.
+ * When you finish writing it, it should solve the "repair the quad"
+ * problem from Assignment 1.  In addition to editing the program,
+ * you should be sure to edit this comment so that it no longer
+ * indicates that the program does nothing.
+ */
+
+import stanford.karel.*;
+
+public class StoneMasonKarel extends SuperKarel {
+
+	// You fill in this part
+	public void run() {
+		
+		turnLeft(); 
+		
+		while (frontIsClear()) {
+			if (beepersPresent()) {
+				move();
+			} else {
+				putBeeper();
+			}
+		}
+		
+		turnRight();
+		
+		for (int i = 0; i < 4; i++) {
+			move();
+		}
+		
+		turnRight();
+		
+		while (frontIsClear()) {
+			if (beepersPresent()) {
+				move();
+			} else {
+				putBeeper();
+			}
+		}
+		

```

Comment:
One of the things that makes you know that you need to modify your code is the fact that you have the same code repeated again and again. In this case that code is:

```
while (frontIsClear()) {
    if (beepersPresent()) {
        move();
    } else {
 	putBeeper();
    }
}
```

Those seven lines could be moved to it's own method.